lib/OpenQA/Worker/Engines/isotovideo.pm

```diff
@@ -230,11 +230,13 @@ sub sync_tests {
             log_info('Finished to rsync tests', channels => 'autoinst');
             last;
         }
-        elsif ($remaining_tries > 1 && $result eq 'exit code 24') {
-            log_info("Rsync failed due to vanished source files ($result), trying again", channels => 'autoinst');
+        elsif ($remaining_tries > 1 && ($result eq 'exit code 24' || $result eq 'exit code 23')) {
+            log_info("Partial transfer due to error ($result), trying again", channels => 'autoinst');
```

Comment:
But now there's the same problem for exit code 24. Either only emit a generic error message or emit 2 distinct error messages. If you want to emit 2 distinct error messages, I'd save them into a hash to avoid having an extra branch within the if. (The hash keys could be the strings you're currently comparing to and hence the hash could also serve to simplify the condition.)

Reply:
Agreed, i already want to add more codes and error messages. 😉